scripts/demo_streamlit.py

```diff
@@ -0,0 +1,67 @@
+import streamlit as st
+import altair as alt
+import polars as pl
+import nisapi
+
+
+def widget_filter(
+    df: pl.DataFrame,
+    column: str,
+    options=None,
+    default: str = None,
+    n_radio_max: int = 5,
+) -> pl.DataFrame:
+    if options is None:
+        options = df[column].unique().to_list()
+
+    if len(options) <= n_radio_max:
+        if default is not None and default in options:
+            options = [default] + [x for x in options if x != default]
+
+        value = st.radio(label=column, options=options)
+    else:
+        if default is not None and default in options:
+            index = options.index(default)
+        else:
+            index = 0
+
+        value = st.selectbox(label=column, options=options, index=index)
+
+    return df.filter(pl.col(column) == pl.lit(value)).drop(column)
+
+
+if __name__ == "__main__":
+    st.title("Locally cached NIS data")
+
+    nis = nisapi.get_nis().collect()
+
+    data = (
+        nis.pipe(widget_filter, "vaccine", default="flu")
+        .pipe(widget_filter, "geographic_type", default="nation")
+        .pipe(widget_filter, "geographic_value")
+        .pipe(widget_filter, "demographic_type", default="overall")
+        .pipe(widget_filter, "demographic_value")
+        .pipe(widget_filter, "time_type", default="week")
+        .pipe(widget_filter, "indicator_type", default="4-level vaccination and intent")
+        .pipe(widget_filter, "indicator_value", default="received a vaccination")
+    )
+
+    datasets = data["id"].unique().to_list()
+
+    st.text(f"Showing data from datasets: {datasets}")
+
+    time_axis = alt.Axis(format="%Y-%b-%d", tickCount="month")
+    points = (
+        alt.Chart(data)
+        .mark_point()
+        .encode(x=alt.X("time_end", axis=time_axis), y="estimate")
+    )
+    error_bars = (
+        alt.Chart(data)
+        .mark_errorbar()
+        .encode(x=alt.X("time_end", axis=time_axis), y="lci", y2="uci")
+    )
+
+    chart = points + error_bars
```

Comment:
Does this end up with the y-axis label being "lci"?

Reply:
Good catch! Fixed in https://github.com/CDCgov/nis-py-api/pull/43/commits/e61816b95f5237ca542c3ca5cd9a307e52c17858

And I caught another bug while I was at it; see https://github.com/CDCgov/nis-py-api/pull/43/commits/5e050aaf7a7194964073ea06116fd61ecfd4ce26